packages/dashboard/src/dashboard-actions/annotations.ts

```diff
@@ -0,0 +1,89 @@
+import { XAnnotation } from '@synchro-charts/core';
+import { DashboardConfiguration } from '../types';
+
+export const addXAnnotation = ({
+  dashboardConfiguration,
+  widgetId,
+  annotation,
+}: {
+  dashboardConfiguration: DashboardConfiguration;
+  widgetId: string;
+  annotation: XAnnotation;
+}) => {
+  return dashboardConfiguration.map((widget) => {
+    if (widget.id == widgetId) {
+      let currAnnotations = widget.annotations;
```

Comment:
is there a way to write this without using a `let`?

Reply:
Yes, refactored to fit this scheme